plugins/reference-assets/core/src/assets/input/__tests__/transform.test.ts

```diff
@@ -0,0 +1,112 @@
+import { runTransform } from "@player-ui/asset-testing-library";
+import { describe, expect, it } from "vitest";
+import { Flow } from "@player-ui/player";
+import { CommonTypesPlugin } from "@player-ui/common-types-plugin";
+import { inputTransform } from "../transform";
+
+describe("Input Asset Transform", () => {
+  it("should transform", () => {
+    const testFlow: Flow = {
+      id: "generated-flow",
+      views: [
+        {
+          id: "input-1",
+          type: "input",
+          label: {
+            asset: {
+              id: "input-1-label",
+              type: "text",
+              value: "Input with validation and formatting",
+            },
+          },
+          note: {
+            asset: {
+              id: "input-1-note",
+              type: "text",
+              value: "It expects a positive integer",
+            },
+          },
+          binding: "foo.bar",
+        },
+      ],
+      schema: {
+        ROOT: {
+          foo: {
+            type: "FooType",
+          },
+        },
+        FooType: {
+          bar: {
+            type: "IntegerPosType",
+            validation: [
+              {
+                type: "required",
+              },
+            ],
+          },
+        },
+      },
+    };
+    const { current } = runTransform("input", inputTransform, testFlow, [
+      new CommonTypesPlugin(),
+    ]);
+    expect(current?.validation).toBeUndefined();
+    expect(current?.dataType?.type).toBe("IntegerPosType");
+  });
+  it("should transform with validation", () => {
+    const testFlow: Flow = {
+      id: "generated-flow",
+      views: [
+        {
+          id: "input-1",
+          type: "input",
+          label: {
+            asset: {
+              id: "input-1-label",
+              type: "text",
+              value: "Input with validation and formatting",
+            },
+          },
+          note: {
+            asset: {
+              id: "input-1-note",
+              type: "text",
+              value: "It expects a positive integer",
+            },
+          },
+          binding: "foo.bar",
+        },
+      ],
+      data: {},
+      schema: {
+        ROOT: {
+          foo: {
+            type: "FooType",
+          },
+        },
+        FooType: {
+          bar: {
+            type: "IntegerPosType",
+            validation: [
+              {
+                type: "required",
+                trigger: "load",
+              },
+            ],
+          },
+        },
+      },
+    };
+    const { current } = runTransform("input", inputTransform, testFlow, [
+      new CommonTypesPlugin(),
+    ]);
+
+    expect(current?.validation).not.toBeUndefined();
```

Comment:
Do we need this if we are already testing for the object `toHaveProperty("severity", "error");` on line 106?

Reply:
I think it just makes the test more easily identifiable if it fails, this will fail for the object not being defined, instead of saying it cant find a property